File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, initLost, initReserve) {
    var answer = 0;
    const lostSet = new Set(initLost);
    const reserveSet = new Set(initReserve);
    
    const lost = Array.from(new Set([...lostSet].filter(x => !reserveSet.has(x))));
    const reserve = Array.from(new Set([...reserveSet].filter(x => !lostSet.has(x))));
    
    let students = Array(n).fill(1)
    // 왼쪽에서부터 점검해주면 된다.
    for (idx = 0; idx < lost.length; idx++) {
        students[lost[idx] - 1] -= 1
    }
    for (idx = 0; idx < reserve.length; idx ++) {
        students[reserve[idx] - 1] += 1
    }
    
    for (idx = 0; idx < n; idx++) {
        if (students[idx] == 0) {
            if (idx > 0 && students[idx - 1] >= 2) {
                students[idx] += 1
                students[idx - 1] -= 1
            } else if (idx < n - 1 && students[idx + 1] >= 2) {
                students[idx] += 1
                students[idx + 1] -= 1
            }
        }
    }
    console.log(students)
    for (idx = 0; idx < n; idx++) {
        if (students[idx]) {
            answer += 1
        }
    }
    return answer;
}

solution(4, [2,3], [3,4])

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    r_lost = list(set(lost) - set(reserve))
    r_reserve = list(set(reserve) - set(lost))
    lst = [1 for _ in range(n)]
    for idx in r_lost:
        lst[idx - 1] -= 1
    for idx in r_reserve:
        lst[idx - 1] += 1
    for idx in range(n):
        if lst[idx] == 0:
            if idx > 0 and lst[idx - 1] == 2:
                lst[idx - 1] -= 1
                lst[idx] += 1
            elif idx < n - 1 and lst[idx + 1] == 2:
                lst[idx + 1] -= 1
                lst[idx] += 1
    print(lst)
    for l in lst:
        if l > 0:
            answer += 1
    return answer